google/resource-snippets/iam-v1/configurable_service_accounts.jinja

```diff
@@ -0,0 +1,33 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resources:
+#- type: iam.v1.serviceAccount
+- type: gcp-types/iam-v1:projects.serviceAccounts
+  name: test-account
+  properties:
+    accountId: crispin-{{ properties['accountIdSuffix'] }}
+    displayName: {{ properties['displayName'] }}
+#- type: iam.v1.serviceAccounts.key
+- type: gcp-types/iam-v1:projects.serviceAccounts.keys
+  name: test-account-key
+  properties:
+    parent: $(ref.test-account.name)
+    privateKeyType: TYPE_GOOGLE_CREDENTIALS_FILE
+- action: gcp-types/iam-v1:iam.projects.serviceAccounts.testIamPermissions
+  name: test-iam-permissions
+  properties:
+    resource: projects/{{ env["project"] }}/serviceAccounts/$(ref.test-account-key.privateKeyData.base64Decode().load().client_email)
```

Comment:
How does one get access to this `base64Decode()` function? I'm getting the following error message on my template:

```
$ gcloud alpha deployment-manager deployments update test-serviceaccount --template=serviceaccount_key.jinja
The fingerprint of the deployment is JOpP6d315IfO_OnO8WiX-g==
Waiting for update [operation-1522263935086-5687db11aedb1-0500754c-3e22b0ef]...failed.
ERROR: (gcloud.alpha.deployment-manager.deployments.update) Error in Operation [operation-1522263935086-5687db11aedb1-0500754c-3e22b0ef]: errors:
- code: REFERENCE_VALUE_NOT_FOUND
  location: /deployments/test-serviceaccount/resources/sa-test-3-key->$.properties
  message: |-
    The reference 'privateKeyData.base64Decode()' is not found, reason:
    The resource 'sa-test-3-key' exists, but the reference value does not, details: Unrecognized function base64Decode
```

```
resources:
  - name: serviceaccount-test-1
    type: iam.v1.serviceAccount
    properties:
      accountId: serviceaccount-test-1
      displayName: serviceaccount-test-1

  - name: sa-test-3-key
    type: iam.v1.serviceAccounts.key
    properties:
      parent: $(ref.serviceaccount-test-1.name)
      privateKeyType: TYPE_GOOGLE_CREDENTIALS_FILE

outputs:
  - name: private_key
    value: $(ref.sa-test-3-key.privateKeyData.base64Decode())
```